### PR TITLE
Document configuration changes in configuration basics

### PR DESCRIPTION
**Pull request by pounard**
First PR for the documentation overhaul, it focuses on the actual *Bundle configuration*, makes it more generic, adds the flavor alternatives, and fixes examples following the new configuration changes from #191

### CODE REVIEW

File: docs/content/configuration/basics.md [hunk 1]
```diff
@@ -0,0 +1,650 @@
...
+
+Configuration options will vary depending on which flavor you want to use.
+
+Select below your target:
+
+<FlavorSwitcher />
+
+<div class="symfony">
+
+The *DbToolsBundle* let you configure some of its behaviors. As with any classic Symfony Bundle,
```
Comment (SimonMellerin) [inline docs/content/configuration/basics.md:11]:
```suggestion
*DbToolsBundle* let you configure some of its behaviors. As with any classic Symfony Bundle,
```

File: docs/content/configuration/basics.md [hunk 2]
```diff
@@ -0,0 +1,650 @@
...
+
+:::tip
+A complete example of this file can be found in the bundle sources in: `vendor/makinacorpus/db-tools-bundle/config/packages/db_tools.yaml`.
+:::
+
+</div>
+
+<div class="standalone">
+
+The *DbToolsBundle* let you configure some of its behaviors. As with any classic Symfony Bundle,
```
Comment (SimonMellerin) [inline docs/content/configuration/basics.md:22]:
```suggestion
*DbToolsBundle* let you configure some of its behaviors, ```

File: docs/content/configuration/basics.md [hunk 3]
```diff
@@ -0,0 +1,650 @@
...
+
+The `db_tools.storage.root_dir` parameter let you choose where to put the generated dumps.
+
+<div class="symfony">
+
+Default value is `'%kernel.project_dir%/var/db_tools'`.
+
+</div>
+
+<div class="symfony">
```
Comment (SimonMellerin) [inline docs/content/configuration/basics.md:101]:
```suggestion
<div class="standalone">
```

File: docs/content/configuration/basics.md [hunk 4]
```diff
@@ -0,0 +1,650 @@
...
+
+Default value is `./var/db_tools'`.
+
+</div>
+
+#### File and directory naming strategy
+
+Default behavior will store your backup using this strategy:
+`%db_tools.storage.root_dir%/<YEAR>/<MONTH>/<CONNECTION-NAME>-<YEAR><MONTH><DAY><HOUR><MINUTES><SECOND>.<EXT>`
+where `<EXT>` is the file extension depending upon the database backend (mostly `.sql` or `.dump`).
```
Comment (SimonMellerin) [inline docs/content/configuration/basics.md:111]: May be to keep a consistency in the doc, we can keep the term 'vendor'

```suggestion
where `<EXT>` is the file extension depending upon the database vendor (mostly `.sql` or `.dump`).
```

File: docs/content/configuration/basics.md [hunk 5]
```diff
@@ -0,0 +1,650 @@
...
+
+This will allow the restore command to find your backups.
+
+</div>
+
+<div class="standalone">
+
+:::warning
+There is as of now no way to implement a custom filename strategy when using *DbToolsBundle* as a standalone
+CLI tool.
```
Comment (SimonMellerin) [inline docs/content/configuration/basics.md:181]: may something like "Feel free to open an issue if you want the feature"

File: docs/content/configuration/basics.md [hunk 6]
```diff
@@ -0,0 +1,650 @@
...
+:::
+
+</div>
+
```
Comment (SimonMellerin) [inline docs/content/configuration/basics.md:185]: May be we can add an :::info telling people that other FilenameStrategy could be added in the future and that they can ask for them if they have idea or something

File: docs/content/configuration/basics.md [hunk 7]
```diff
@@ -0,0 +1,650 @@
...
+processes timeout in seconds.
+
+Default value is `600` (seconds) for backup, `1800` (seconds) for restore.
+
+Value can be either a [\DateInterval::createFromDateString()](https://www.php.net/manual/en/dateinterval.createfromdatestring.php)
+compatible string value or a number of seconds as an integer value.
+
+<div class="symfony">
+
+Here is an example for exclude `table1` and `table2` for all connections:
```
Comment (SimonMellerin) [inline docs/content/configuration/basics.md:308]: Comment is wrong here

File: docs/content/configuration/basics.md [hunk 8]
```diff
@@ -0,0 +1,650 @@
...
+    # As a date interval string.
+    backup_timeout: '6 minutes 30 seconds'
+    restore_timeout: '3 minutes 15 seconds'
+
+    # As a number of seconds integer value.
+    backup_timeout: 390
+    restore_timeout: 195
+```
+
+Or set a specific table list for each connection:
```
Comment (SimonMellerin) [inline docs/content/configuration/basics.md:322]: Comment is wrong here

File: docs/content/configuration/basics.md [hunk 9]
```diff
@@ -0,0 +1,650 @@
...
+            backup_timeout: '6 minutes 30 seconds'
+            restore_timeout: '3 minutes 15 seconds'
+        connection_two:
+            backup_timeout: 390
+            restore_timeout: 195
+```
+</div>
+<div class="standalone">
+
+Here is an example for exclude `table1` and `table2` for all connections:
```
Comment (SimonMellerin) [inline docs/content/configuration/basics.md:338]: Comment is wrong here

File: docs/content/configuration/basics.md [hunk 10]
```diff
@@ -0,0 +1,650 @@
...
+# As a date interval string.
+backup_timeout: '6 minutes 30 seconds'
+restore_timeout: '3 minutes 15 seconds'
+
+# As a number of seconds integer value.
+backup_timeout: 390
+restore_timeout: 195
+```
+
+Or set a specific table list for each connection:
```
Comment (SimonMellerin) [inline docs/content/configuration/basics.md:352]: Comment is wrong here

File: docs/content/configuration/basics.md [hunk 11]
```diff
@@ -0,0 +1,650 @@
...
+
+`db-tools:backup` and `db-tools:restore` need your system/environment to provide some extra binaries
+to be able to work. These binaries depend on the database vendor you use, you will need:
+* for MariaDB: `mariadb-dump` and `mariadb`
+* for MySQL: `mysqldump` and `mysql`
+* for PostgreSQL: `pg_dump` and `pg_restore`
+* for SQLite: `sqlite3`
+
+You can verify if those binaries are well found by the *DbToolsBundle*,
+for each of your DBAL connections, by launching:
```
Comment (SimonMellerin) [inline docs/content/configuration/basics.md:376]:
```suggestion
for each of your connections, by launching:
```

File: docs/content/configuration/basics.md [hunk 12]
```diff
@@ -0,0 +1,650 @@
...
+</div>
+<div class="standalone">
+
+```sh
+php vendor/bin/db-tools database:check
+```
+</div>
+
+If the `db-tools:check` command returns you some errors:
+ * if your binaries are present on your system but the *DbToolsBundle* can't find them you will need
```
Comment (SimonMellerin) [inline docs/content/configuration/basics.md:392]:
```suggestion
 * if your binaries are present on your system but *DbToolsBundle* can't find them you will need
```

File: docs/content/configuration/basics.md [hunk 13]
```diff
@@ -0,0 +1,650 @@
...
+```
+:::
+
+:::warning
+Dump and restore is not supported yet for SQL Server.
+:::
+
+### Default binary options
+
+Apart from the essential options (credentials, database name, etc.), the bundle
```
Comment (SimonMellerin) [inline docs/content/configuration/basics.md:472]:
```suggestion
Apart from the essential options (credentials, database name, etc.), the library
```

File: docs/content/configuration/basics.md [hunk 14]
```diff
@@ -0,0 +1,650 @@
...
+  * MariaDB: None
+  * MySQL: None
+  * PostgreSQL: `-j 2 --clean --if-exists --disable-triggers`
+  * SQLite: None
+
+## Anonymizer paths
+
+<div class="symfony">
+
+By default, the *DbToolsBundle* will look for custom *anonymizers* in 2 directories:
```
Comment (SimonMellerin) [inline docs/content/configuration/basics.md:544]:
```suggestion
By default, *DbToolsBundle* will look for custom *anonymizers* in 2 directories:
```

File: docs/content/configuration/basics.md [hunk 15]
```diff
@@ -0,0 +1,650 @@
...
+db_tools:
+    anonymizer_paths:
+        - '%kernel.project_dir%/src/Anonymizer'
+        - '%kernel.project_dir%/vendor/myAnonymizerProvider/anonymizers/src'
+        # ...
+```
+</div>
+<div class="standalone">
+
+By default, the *DbToolsBundle* when used as a standalone CLI tool will not lookup for
```
Comment (SimonMellerin) [inline docs/content/configuration/basics.md:564]:
```suggestion
By default, *DbToolsBundle* when used as a standalone CLI tool will not lookup for
```

File: docs/content/configuration/basics.md [hunk 16]
```diff
@@ -0,0 +1,650 @@
...
+custom *anonymizers*.
+
+If you want to write custom anonymizers, in order for them to be found, you must specify
+paths where the source code lies:
```
Comment (SimonMellerin) [inline docs/content/configuration/basics.md:568]:
```suggestion
paths where the source code lives:
```

File: docs/content/configuration/basics.md [hunk 17]
```diff
@@ -0,0 +1,650 @@
...
+:::warning
+Packs must be installed using composer: *DbToolsBundle* uses composer generated metadata
+about installed packages to find them.
+:::
+
+## Anonymization
+
+<div class="symfony">
+
+Per default, the **DbToolsBundle** will only look for anonymization configurations from PHP attributes on Doctrine Entities.
```
Comment (SimonMellerin) [inline docs/content/configuration/basics.md:595]:
```suggestion
Per default, **DbToolsBundle** will only look for anonymization configurations from PHP attributes on Doctrine Entities.
```

File: docs/content/configuration/basics.md [hunk 18]
```diff
@@ -0,0 +1,650 @@
...
+
+But *DbToolsBundle* does not necessary need Doctrine ORM to anonymize your data, it can do it just with a DBAL connection.
+In this case (or if you prefer YAML over attributes): you can configure the *DbToolsBundle* to look for anonymization
```
Comment (SimonMellerin) [inline docs/content/configuration/basics.md:598]:
```suggestion
In this case (or if you prefer YAML over attributes): you can configure *DbToolsBundle* to look for anonymization
```

File: docs/content/configuration/basics.md [hunk 19]
```diff
@@ -0,0 +1,650 @@
...
+        connection_one:
+            - '%kernel.project_dir%/config/anonymizations/connection_one_1.yaml'
+            - '%kernel.project_dir%/config/anonymizations/connection_one_2.yaml'
+        # ...
+```
+
+</div>
+<div class="standalone">
+
+You need to register your anonymization configuration for the anonymization feature to work:
```
Comment (SimonMellerin) [inline docs/content/configuration/basics.md:624]: May be you can start with the simple version with anonymization directly in the config file, and then tell that one can choose to extract this conf to another file

**Review by SimonMellerin — CHANGES_REQUESTED** [review]
First review only on `basics.md`

File: docs/content/configuration/basics.md (hunk 16 above)
Comment (pounard) [reply docs/content/configuration/basics.md:568]: This is valid sentences using the verb "to lie", but if you prefer I can change it ?

Comment (SimonMellerin) [reply docs/content/configuration/basics.md:568]: ah ok, my bad, keep it

File: docs/content/configuration/basics.md (hunk 5 above)
Comment (pounard) [reply docs/content/configuration/basics.md:181]: I added:

> If you need this feature, please let us know by [filing an issue](https://github.com/makinacorpus/DbToolsBundle/issues).

Is it OK ?

File: docs/content/configuration/basics.md (hunk 6 above)
Comment (pounard) [reply docs/content/configuration/basics.md:185]:
```
:::info
More filename strategies may be implemented in core in the future. If you have any
suggestions, please [open an discussion](https://github.com/makinacorpus/DbToolsBundle/issues) about it.
:::
```

Is good ?

File: docs/content/configuration/basics.md (hunk 19 above)
Comment (pounard) [reply docs/content/configuration/basics.md:624]: I was thinking that this part was not right. Your suggestion seems good.

**Review by SimonMellerin — APPROVED** [review]

**Review by SimonMellerin — APPROVED** [review]

**Pull request merged**